addons/sourcemod/scripting/gokz-quiet/hideplayers.sp

```diff
@@ -0,0 +1,305 @@
+
+	// Lateload support
+	for (int client = 1; client <= MaxClients; client++)
+	{
+		if (IsValidClient(client))
+		{
+			OnJoinTeam_HidePlayers(client, GetClientTeam(client));
+		}
+	}
+}
+
+void OnJoinTeam_HidePlayers(int client, int team)
+{
+	// Make sure client is only ever hooked once
+	SDKUnhook(client, SDKHook_SetTransmit, OnSetTransmitClient);
+	
+	if (team == CS_TEAM_T || team == CS_TEAM_CT)
+	{
+		SDKHook(client, SDKHook_SetTransmit, OnSetTransmitClient);
+	}
+}
+
+Action CommandToggleShowPlayers(int client, int args)
+{
+	if (GOKZ_GetOption(client, gC_QTOptionNames[QTOption_ShowPlayers]) == ShowPlayers_Disabled)
+	{
+		GOKZ_SetOption(client, gC_QTOptionNames[QTOption_ShowPlayers], ShowPlayers_Enabled);
+	}
+	else
+	{
+		GOKZ_SetOption(client, gC_QTOptionNames[QTOption_ShowPlayers], ShowPlayers_Disabled);
+	}
+	return Plugin_Handled;
+}
+
+// =====[ PRIVATE ]=====
+
+// Hide most of the other players' actions. This function is expensive.
+static Action OnSetTransmitClient(int entity, int client)
+{
+	if (GOKZ_GetOption(client, gC_QTOptionNames[QTOption_ShowPlayers]) == ShowPlayers_Disabled
+		 && entity != client
+		 && entity != GetObserverTarget(client))
+	{
+		return Plugin_Handled;
+	}
+	return Plugin_Continue;
+}
+
+// Hide reload sounds. Required if other players were visible at one point during the gameplay.
+static Action Hook_WeaponSound(UserMsg msg_id, Protobuf msg, const int[] players, int playersNum, bool reliable, bool init)
+{
+	int newClients[MAXPLAYERS], newTotal = 0;
+	int entidx = msg.ReadInt("entidx");
+	for (int i = 0; i < playersNum; i++)
+	{
+		int client = players[i];
+		if (GOKZ_GetOption(client, gC_QTOptionNames[QTOption_ShowPlayers]) == ShowPlayers_Enabled
+			|| entidx == client
+			|| entidx == GetObserverTarget(client))
+		{
+			newClients[newTotal] = client;
+			newTotal++;
+		}
+	}
+
+	// Nothing's changed, let the engine handle it.
+	if (newTotal == playersNum)
+	{
+		return Plugin_Continue;
+	}
+
+	// Only way to modify the recipient list is to RequestFrame and create our own user message.
+	char path[PLATFORM_MAX_PATH];
+	msg.ReadString("sound", path, sizeof(path));
+	int flags = USERMSG_BLOCKHOOKS;
+	if (reliable)
+	{
+		flags |= USERMSG_RELIABLE;
+	}
+	if (init)
+	{
+		flags |= USERMSG_INITMSG;
+	}
+
+	DataPack dp = new DataPack();
+	dp.WriteCell(msg_id);
+	dp.WriteCell(newTotal);
+	dp.WriteCellArray(newClients, newTotal);
+	dp.WriteCell(flags);
+	dp.WriteCell(entidx);
+	dp.WriteFloat(msg.ReadFloat("origin_x"));
+	dp.WriteFloat(msg.ReadFloat("origin_y"));
+	dp.WriteFloat(msg.ReadFloat("origin_z"));
+	dp.WriteString(path);
+	dp.WriteFloat(msg.ReadFloat("timestamp"));
+
+	RequestFrame(RequestFrame_WeaponSound, dp);
+	return Plugin_Handled;
+}
+
+static void RequestFrame_WeaponSound(DataPack dp)
+{
+	dp.Reset();
+
+	UserMsg msg_id = dp.ReadCell();
+	int newTotal = dp.ReadCell();
+	int newClients[MAXPLAYERS];	
+	dp.ReadCellArray(newClients, newTotal);

```

Comment:
Could you append a fix on here in this pr? I tested it on my server(only 1 player online) and error throws
[SM] Exception reported: Invalid data pack operation: current position isn't an array!

**DataPack doesn't support CellArray or FloatArray with size 0**
**You have to jump to the next postition**

Reply:
That's strange. The reloading client should always be included in the array, which makes  its size 1 at minimum...

Reply:
Not a listen server.
It seems that this error throws **only with 1 player and GOTV** on the server
I removed GOTV and went fine, maybe it's just a little probleme:eyes:.

Reply:
i also tested that the Callback param `playersNum` **always equal to 0** if only 1 player on the server

Reply:
if `playersNum` is 0, then that means `newTotal` must be 0 as well, and the plugin will return without ever calling `RequestFrame`, so it shouldn't be throwing an exception, right?